akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/TimerScheduler.scala

```diff
@@ -53,11 +53,20 @@ trait TimerScheduler[T] {
    * for the same key, will not be received by the actor, even though the message might already
    * be enqueued in the mailbox when cancel is called.
    */
-  def cancel(key: Any): Unit
+  def cancel(key: K): Unit
 
   /**
    * Cancel all timers.
    */
   def cancelAll(): Unit
 
 }
+
+trait TimerScheduler[T] extends KeyTypedTimerScheduler[Any, T] {
+
+  /**
+   * Specialises the type of the key to use for the timers.
+   */
+  def withKeyType[K]: KeyTypedTimerScheduler[K, T] = this
```

Comment:
Let's see what kind of problem we are trying to solve here. The type on the key could give some additional safety to not cancel or replace wrong.

```
Behaviors.withTimers { timers =>
  timers.startSingleTimer("timer", TimeIsUp, 5.seconds)
  ...
}

// elsewhere, using wrong key
timers.cancel(TimeIsUp)
```

I'm not convinced that is a big problem. More common is probably to make a typo when being sloppy and not defining the keys as constants.

```
timers.cancel("timmers")
```

Also, I don't think many users will find this extra `withKeyType` method, and it makes it look more complicated than it is if we sprinkle that usage in the docs.

Reply:
I'm happy to close this until more motivation arises.

I'll submit the underlying improvements.